rust/protocol/src/session_cipher.rs

```diff
@@ -425,8 +425,12 @@ fn decrypt_message_with_record<R: Rng + CryptoRng>(
                 previous_state_count(),
             );
         }
-        Err(SignalProtocolError::MessageDecryptionFailed(
-            create_decryption_failure_log(remote_address, &errs, record, ciphertext)?,
+        log::error!(
+            "{}",
+            create_decryption_failure_log(remote_address, &errs, record, ciphertext)?
```

Comment:
Can we go even further and have `create_decryption_failure_log` just call `log::error!` multiple times? Or is it worth having it all on one line?

Reply:
I considered that. But the list of possible chains is IMO one "log", not something to be considered individually. OTOH they all will get dumped consecutively into the file anyway so either way is probably fine? IDK, no strong feelings here from me. And I don't think Android cares either way as long as the bits end up in the debug log somehow.